Canary test is consistently failing to verify csi-addons sidecar

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Two canary tests (`canary` and `nvmeof-protocol`) are failing consistently since the merge of #17599, it appears related to changing the csi replicas to 2. Likely the parsing is confused by finding two pods instead of one.
```
rook-ceph.rbd.csi.ceph.com-ctrlplugin-8c8c4d89-xqcmb' '-o=jsonpath={.status.containerStatuses[?(@.name=='\''csi-addons'\'')].ready}'
+ container_status=
+ '[' '' '!=' true ']'
+ echo 'csi-addons container not found in rook-ceph.rbd.csi.ceph.com-ctrlplugin-8c8c4d89-d9fx2
rook-ceph.rbd.csi.ceph.com-ctrlplugin-8c8c4d89-xqcmb pod!'
+ exit 1
csi-addons container not found in rook-ceph.rbd.csi.ceph.com-ctrlplugin-8c8c4d89-d9fx2
rook-ceph.rbd.csi.ceph.com-ctrlplugin-8c8c4d89-xqcmb pod!
Error: Process completed with exit code 1.
```

One of the csi controllers is pending, so we just need to change the replicas back to 1 in the tests.
```
rook-ceph.cephfs.csi.ceph.com-ctrlplugin-754b7895f7-qxczc   6/6     Running   0          89s
rook-ceph.cephfs.csi.ceph.com-ctrlplugin-754b7895f7-tc8x8   0/6     Pending   0          89s
rook-ceph.rbd.csi.ceph.com-ctrlplugin-8c8c4d89-5h4kd        6/6     Running   0          89s
rook-ceph.rbd.csi.ceph.com-ctrlplugin-8c8c4d89-6scqg        0/6     Pending   0          89s
```

**Expected behavior:**
Canary tests should pass consistently.

--- a/tests/scripts/github-action-helper.sh
+++ b/tests/scripts/github-action-helper.sh
@@ -305,6 +305,9 @@ function remove_cluster_prerequisites() {
 }
 
 function deploy_manifest_with_local_build() {
+  if [[ "$1" == "deploy/examples/operator.yaml" || "$1" == "operator.yaml" ]]; then
+     sed -i "s|replicas: 2|replicas: 1|g" $1
+  fi
   if [[ "$USE_LOCAL_BUILD" != "false" ]]; then
     sed -i "s|image: docker.io/rook/ceph:.*|image: docker.io/rook/ceph:local-build|g" $1
   fi

